src/index.js

```diff
@@ -51,24 +68,31 @@ function syncReduxAndRouter(history, store, selectRouterState = SELECT_STATE) {
   }
 
   const unsubscribeHistory = history.listen(location => {
-    const routePath = locationToString(location);
+    const route = {
+      path: history.createPath(location),
+      state: location.state
+    };
+
+    // Avoid dispatching an action if the store is already up-to-date,
+    // even if `history` wouldn't do anything if the location is the same
+    if(locationsAreEqual(getRouterState(), route)) return;
```

Comment:
Yeah, good idea. I got some failing tests when just removing it, so I'll try to take a look at it and find a solution